server/core/src/main/java/org/jolokia/server/core/util/NetworkInterfaceAndAddress.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2009-2025 Roland Huss
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jolokia.server.core.util;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+
+public class NetworkInterfaceAndAddress {
+
+    public final NetworkInterface networkInterface;
+    public final InetAddress address;
+
+    public NetworkInterfaceAndAddress(NetworkInterface networkInterface, InetAddress address) {
+        this.networkInterface = networkInterface;
+        this.address = address;
+    }
+
+}
```

Comment:
Time to consider introducing Java record? :-)  Just an idea...

Reply:
Hold your horses - we've only recently moved away from Java 8 ;)
But I get your point - would be great!